models.ts

```diff
@@ -8,11 +11,51 @@ class Peli {
 }
 
 class PelisCollection {
-  getAll(): Promise<Peli[]> {
-    return jsonfile("...laRutaDelArchivo").then(() => {
-      // la respuesta de la promesa
-      return [];
+  pelis :  Peli[]
+  getAll() {
+    const promesa = jsonfile.readFile("./pelis.json")
+     promesa.then((res) => {
+      this.pelis = res
+      return this.pelis
     });
+     return promesa
+ }
+   getById(id:number){
+    return  this.getAll()
+    .then( respuesta =>
+      respuesta.find((p)=>{
+        if(p.id == id){
+          return p
+        }
+      })) 
+    } 
+
+    search(options: any) {
+      return this.getAll()
+      .then((res) => {
+         let respuesta = res
+
+         if (options.hasOwnProperty("title")) {
+          respuesta = res.filter((p) => p.title.includes(options.title))
+         }
+         if(options.hasOwnProperty("tag")){
+          respuesta = filtrar(respuesta, function(item){return item.tags.includes(options.tag)})
+         }
+         return respuesta
+      });
+   }
+   add(peli: Peli): Promise<any> {
+      return this.getAll().then((json) => {
+        let repeatedId = find(json,function(item) {return item.id === peli.id});
```

Comment:
Esta lógica la resolves en el getById, la podés reutilizar haciendo this.getById